src/main/scala/sbtcrossproject/CrossProject.scala

```diff
@@ -1,184 +1,129 @@
-   */
-  def settingSets(select: AddSettings*): CrossProject =
-    mapProjects(_.settingSets(select: _*))
+  def overrideConfigs(cs: Configuration*): CrossProject[P] =
+    copy(root, projects.map((x) => x._1 -> x._2.overrideConfigs(cs: _*)))
 
-  def settings(ss: Def.Setting[_]*): CrossProject =
-    mapProjects(_.settings(ss: _*))
+  def dependsOn(deps: CrossClasspathDependency*): CrossProject[P] =
+    copy(
+        root,
+        projects.map((x) => x._1 -> x._2.dependsOn(deps.map(_.dep(x._1)): _*)))
 
-  override def toString(): String = {
+  override def toString: String = {
     val entries = this.projects.map {
-      case (platform, project) => "$platform = $project"
+      case (t, p) => s"${t.name} -> $p"
     }
-    s"CrossProject(${entries.mkString(", ")})"
+    s"CrossProject(${entries.mkString(",\n")})"
   }
 
-  // Helpers
-
-  private def mapCrossPlatforms(transform: CrossPlatform => Project): CrossProject =
-    copy(projects.map { case (p, _) => p -> transform(p) })
+  private[sbtcrossproject] def settings(
+      t: Platform,
+      settings: Seq[_root_.sbt.Def.Setting[_]]): CrossProject[P] =
+    copy(root, projects + (t -> projects(t).settings(settings)))
 
-  private def mapProjects(transform: Project => Project): CrossProject =
-    copy(projects.mapValues(transform))
+  private[sbtcrossproject] def path(t: Platform,
+                                    path: String): CrossProject[P] = {
+    copy(root, projects + (t -> projects(t).in(root / path)))
+  }
 
-  private def copy(projects: Map[CrossPlatform, Project] = projects): CrossProject =
-    new CrossProject(crossType, projects)
 }
 
 object CrossProject extends CrossProjectExtra {
-  def apply(id: String, platforms: Seq[CrossPlatform], base: File,
-      crossType: CrossType): CrossProject =
-    apply(platforms.map(p => (p, id + p.name)).toMap, base, crossType)
 
-  def apply(ids: Map[CrossPlatform, String], base: File,
-      crossType: CrossType): CrossProject = {
+  def crossProject_impl(c: Context): c.Expr[Builder[Platform]] = {
+    import c.universe._
+    val enclosingValName = MacroUtils.definingValName(
+        c,
+        methodName =>
+          s"""$methodName must be directly assigned to a val, such as `val x = $methodName`.""")
+    val name = c.Expr[String](Literal(Constant(enclosingValName)))
+    reify {
+      new Builder(name.splice, file("."), Map.empty, CrossType.Full)
+    }
+  }
 
-    val sss = sharedSrcSettings(crossType)
+  implicit def toCrossProject[P <: Platform](
+      builder: Builder[P]): CrossProject[P] = {
 
-    val projects = ids.map { case (p, id) =>
-      val project = Project(id, crossType.dir(p, base)).
-        settings(sss: _*).
-        enablePlugins(p.plugin.toSeq: _*)
+    unmanagedSourceDirectories in Compile ++= builder.crossType
+      .srcPaths(builder.root, builder.map)
 
-      p -> project
-    }
+    unmanagedSourceDirectories in Test ++= builder.crossType
+      .testPaths(builder.root, builder.map)
```

Comment:
I think that, at this point, you need to start testing more seriously than what you are doing now. These settings don't do anything, because they're never passed to `.settings()` (recall that settings are just immutable values of some data type), so these two lines cannot work.